[FIX] portal: postmerge layout fixes

chgo-odoo opened the pull request:
# Portal layout fixes

This PR fixes some issues within Portal, mainly about spacing, layout, and font-size.

**Requires**: https://github.com/odoo/enterprise/pull/53383

## Why do we introduce these changes
We recently merged the portal redesign. While the redesign is almost done for every screen, we needed to fine tune some spacing or fix some layout issues.

## List of the different changes

### portal:

- [x] Review the unconsistent use of hr tag
- [x] Improve information architecture by fixing the typescale of the subtitles
- [x] Remove the subtitle above the products table, improving the design of the invoice
- [x] Fix an issue about the number of messages in the chatter missing a space

task-3651084

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/147496).

chgo-odoo commented:
This PR should also include a fix for https://github.com/odoo/odoo/issues/147801

chgo-odoo commented:
Hi @xlu-odoo, could you have a look at this PR and its Enterprise counterpart, please?

Thanks !

xlu-odoo commented on addons/portal/views/portal_templates.xml:
```diff
@@ -282,7 +282,7 @@
     <template id="portal_record_sidebar" name="My Portal Record Sidebar">
         <div t-attf-class="#{classes}">
-            <div class="o_portal_sidebar_content sticky-lg-top d-lg-inline-block mb-4 mb-lg-0 p-3 p-lg-0" id="sidebar_content">
+            <div class="o_portal_sidebar_content sticky-lg-top d-lg-inline-block mb-4 mb-lg-0 mt-lg-n5 p-3 p-lg-0 pt-lg-5" id="sidebar_content">
```
While it's a nice improvement, price is now touching the header, and needs more spacing. 
Maybe it needs some custo here in my opinion.

chgo-odoo commented:
Thanks @xlu-odoo, I've pushed the changes in a fixup on top of the branch, could you check?

xlu-odoo commented on addons/portal/static/src/scss/portal.scss:
```diff
@@ -359,6 +359,13 @@ img, .media_iframe_video, .o_image {
 
     .o_portal_sidebar_content {
         z-index: -1;
+        padding: map-get($spacers, 3);
+
+        @include media-breakpoint-up(lg) {
+            margin-top: $spacer * -4;
+            padding: 0;
+            padding-top : $spacer * 4;
+        }
```
I guess it's OK. 
Maybe it needs some js magic to calculate the header/navbar height.

Feyensv reviewed (approved):
lgtm, but some .pot updates are missing :eyes:

chgo-odoo commented:
Hi @Feyensv, thanks for having a look that quick !

I've addressed the .po files change in https://github.com/odoo/odoo/pull/147496/commits/9f99234ec91feb3f53eb7e08b985d1977ed101e3. I already encountered some PR that didn't need a manual update of the .po files so I'm not that experienced with this kind of changes. Could you eventually double check the diff, please?

I'm also wondering something. I often see that the file try to target a specific element in the DOM (e.g a strong tag, a title, ...) Let's say I changed a <h1> tag for a <h2>, or I removed/added something from the original selector, does it have to be updated to here or is the file dynamic and able to find the selector even with my changes on it?

Thanks !

Feyensv commented:
> I've addressed the .po files change in [9f99234](https://github.com/odoo/odoo/commit/9f99234ec91feb3f53eb7e08b985d1977ed101e3). I already encountered some PR that didn't need a manual update of the .po files so I'm not that experienced with this kind of changes. Could you eventually double check the diff, please?

You should never update `.po` files yourself, what you should update is the `.pot` file, by reexporting the translations through the "Export Translations" menu/wizard in a database.  `.po` files are automatically generated by Transifex, based on the `.pot` files and updated every week? on odoo by the bot C3POdoo.

> I'm also wondering something. I often see that the file try to target a specific element in the DOM (e.g a strong tag, a title, ...) Let's say I changed a `h1` tag for a `h2`, or I removed/added something from the original selector, does it have to be updated to here or is the file dynamic and able to find the selector even with my changes on it?

Depends on where it's done, AFAIK for standard views, xml/html content is skipped, only raw strings inside the nodes are exported. But for some specific cases, like mail templates, the whole template is exported to be translated at once, and it contains the qweb/html/xml content, therefore any change in those mail templates, even if you touch seemingly untranslatable content, you have to re-export the `.pot` (and be aware that you are breaking all existing translations of the template).

chgo-odoo commented:
Thanks for pointing out the documentation and for your explanation !
I've re-exported both files in https://github.com/odoo/odoo/pull/147496/commits/f35b1c9bd95a9e811e62a12b4704899b486766c7

Feyensv commented on addons/sale/i18n/sale.pot:
```diff
@@ -3374,6 +3366,7 @@ msgstr ""
 #: model:ir.model.fields.selection,name:sale.selection__product_document__attached_on__quotation
 #: model:ir.model.fields.selection,name:sale.selection__sale_order__state__draft
 #: model:ir.model.fields.selection,name:sale.selection__sale_report__state__draft
+#: model:ir.model.fields.selection,name:sale.selection__sale_subscription_report__state__draft
```
This shows that you exported the terms with `sale`and `sale_subscription` installed in your database.
The best way to export translations is when there is only the required module installed ;).

The diff you have in the `.pot` should not be as big and related to your changes.

chgo-odoo commented:
Alright, I've cleared the .pot files to remove changes that are not related to my PR, and exported them with only the two modules installed, we might be good this time 😅

Feyensv commented:
Don't forget `sale_subscription.pot` in your enterprise branch :eyes:

chgo-odoo commented:
Thanks a lot 😬

This time we have everything !

Feyensv commented on addons/sale/views/sale_portal_templates.xml:
```diff
@@ -474,57 +470,52 @@
+                                </div>
+                            </div>
+                            <div class="small d-lg-inline-block">Date: <span class="text-muted" t-field="i.invoice_date"/></div>
+
```
```suggestion
```

Feyensv commented on addons/sale/views/sale_portal_templates.xml:
```diff
@@ -474,57 +470,52 @@
+                                    <span t-out="i.name"/>
+                                </a>
+                                <div t-if="i.payment_state in ('paid', 'in_payment')" class="small badge text-bg-success orders_label_text_align">
+                                <i class="fa fa-fw fa-check"/> Paid
```
```suggestion
                                    <i class="fa fa-fw fa-check"/> Paid
```

Feyensv reviewed (approved):
LGTM for `sale` module changes :+1:

chgo-odoo commented:
Hello @odoo/rd-logistics,

Could you have a look at the small changes related to `sale_stock` ?

Thanks !

xavierbol reviewed (approved):
LGTM for the services part, thanks for your work. 🙂

Feyensv reviewed (approved):
Minor comment I noticed on this last review, to remove in the .pot updates and good to go.

@robodoo delegate+

Feyensv commented on addons/portal/static/src/xml/portal_chatter.xml:
```diff
@@ -6,8 +6,8 @@
                 <span class="o_message_count"> <t t-esc="count"/></span>
-                <t t-if="count == 1">comment</t>
-                <t t-else="">comments</t>
+                <t t-if="count == 1"> comment</t>
+                <t t-else=""> comments</t>
```
Note: probably requires a pot update as well.

chgo-odoo commented:
@robodoo r+

robodoo commented:
@chgo-odoo because this PR has multiple commits, I need to know how to merge it:

* `merge` to merge directly, using the PR as merge commit message
* `rebase-merge` to rebase and merge, using the PR as merge commit message
* `rebase-ff` to rebase and fast-forward

chgo-odoo commented:
@robodoo rebase-ff r+

robodoo commented:
Merge method set to rebase and fast-forward.

Feyensv commented on addons/sale_stock/i18n/sale_stock.pot:
```diff
@@ -4,10 +4,9 @@
 #
 msgid ""
 msgstr ""
-"Project-Id-Version: Odoo Server 17.0\n"
```
Same

chgo-odoo commented on addons/sale/i18n/sale.pot:
```diff
@@ -4,10 +4,9 @@
 #
 msgid ""
 msgstr ""
-"Project-Id-Version: Odoo Server 17.0\n"
```
I think there has been a mistake here, I guess I first remove the line because it had a `+e` at the end and I might have removed the whole line in the end. I can reopen a PR on my side to fix the line if needed.

Feyensv replied:
Please do so (and fix it in the forward-port of this PR since you already have a conflict anyway).
I'm not aware of the real implications of this line in the .pot files, but it could be problematic if it blocked the synchronization from Transifex.
